crypto/src/merkle/proofs.rs

```diff
@@ -255,6 +255,172 @@ impl<H: Hasher> BatchMerkleProof<H> {
         v.remove(&1).ok_or(MerkleTreeError::InvalidProof)
     }
 
+    /// Computes the uncompressed Merkle paths which aggregate to this proof.
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// * No indexes were provided (i.e., `indexes` is an empty slice).
+    /// * Number of provided indexes is greater than 255.
+    pub fn into_paths(&self, indexes: &[usize]) -> Result<Vec<Vec<H::Digest>>, MerkleTreeError> {
```

Comment:
nit: usually, methods which are of the form `into_something()` take ownership of the `self` parameter (rather than take it by reference). If we do this here, we could avoid some cloning in the function.

Reply:
The only place where there is a clone of `self` is in `let mut leaves = self.leaves.clone();`, if this is changed then it will make it necessary to adapt the algorithm, I think, as `self.leaves` is used later, in its original state, in later parts of the code.

Reply:
ah! I don't think we need to change the algorithm then. But I'd still make the function take ownership of `self` as it is consistent with naming convention.

Reply:
Makes sense.